BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBank.swift

```diff
@@ -53,17 +53,24 @@ import GiniCaptureSDK
 
     /**
      Resolves the payment via payment request id.
-
+     **Important**: The amount string in the `PaymentInfo` must be convertible to a `Double`.
+           * For ex. "12.39" is valid, but "12.39 €" or "12,39" are not valid.
      - parameter paymentRequesId: Id of payment request.
      - parameter completion: An action for processing asynchronous data received from the service with Result type as a paramater. Result is a value that represents either a success or a failure, including an associated value in each case.
      Completion block called on main thread.
      In success returns the resolved payment request structure.
      In case of failure error from the server side.
+     If the amount string could not be parsed throws `amountParsingError`
 
      */
     public func resolvePaymentRequest(paymentRequesId: String, paymentInfo: PaymentInfo, completion: @escaping (Result<ResolvedPaymentRequest, GiniBankError>) -> Void) {
-
-        paymentService.resolvePaymentRequest(id: paymentRequesId, recipient: paymentInfo.recipient, iban: paymentInfo.iban, amount: paymentInfo.amount, purpose: paymentInfo.purpose, completion: { result in
+        var amountString = ""
+        do {
+            amountString = try String.parseAmountStringToBackendFormat(string: paymentInfo.amount)
+        } catch {
+            completion(.failure(.amountParsingError(amountString: paymentInfo.amount)))
+        }
+        paymentService.resolvePaymentRequest(id: paymentRequesId, recipient: paymentInfo.recipient, iban: paymentInfo.iban, amount: amountString, purpose: paymentInfo.purpose, completion: { result in
```

Comment:
This line is reached even if the exception was caught and completion was called with failure.

What do you think of this alternative? It forwards the GiniBankError in the completion and warns us if we throw an unexpected error ([assertionFailure()](https://www.swiftbysundell.com/articles/picking-the-right-way-of-failing-in-swift/#lets-start-with-a-list) is ignored in release builds and only affects debug builds):
```
        var amountString: String?
        do {
            amountString = try String.parseAmountStringToBackendFormat(string: paymentInfo.amount)
        } catch let error as GiniBankError {
            completion(.failure(error))
        } catch {
            assertionFailure("Unexpected error: \(error)")
        }
        
        guard let amountString = amountString else {
            return
        }
        
        paymentService.resolvePaymentRequest(id: paymentRequesId, recipient: paymentInfo.recipient, iban: paymentInfo.iban, amount: amountString, purpose: paymentInfo.purpose, completion: { result in
            DispatchQueue.main.async {
                switch result {
                case let .success(resolvedPayment):
                    completion(.success(resolvedPayment))
                case let .failure(error):
                    completion(.failure(.apiError(error)))
                }
            }
        })
```

Reply:
@a-szotyori Looks good, would you like to commit this suggestion?

Reply:
I'll do a change